src/tbb/market.cpp

```diff
@@ -364,22 +363,29 @@ arena* market::arena_in_need ( arena_list_type* arenas, arena* hint ) {
             } while ( arenas[curr_priority_level].empty() );
             it = arenas[curr_priority_level].begin();
         }
-        if( a.num_workers_active() < a.my_num_workers_allotted.load(std::memory_order_relaxed) ) {
+
+        if ( &a == current ) {
+            if ( a.my_pool_state.load(std::memory_order_relaxed) != arena::SNAPSHOT_EMPTY &&
```

Comment:
I am not understand why we check `my_pool_state` for the current arena but we do not check it for other arenas. Overall, why do we have this check? Is not `my_num_workers_allotted` is zero when `arena` is empty?